unity-renderer/Assets/Scripts/MainScripts/DCL/Helpers/Utils/Utils.cs

```diff
@@ -58,6 +59,23 @@ public static void CleanMaterials(Renderer r)
             }
         }
 
+        public static Vector2[] FloatArrayToV2List(RepeatedField<float> uvs)
+        {
+            Vector2[] uvsResult = new Vector2[uvs.Count / 2];
+            int uvsResultIndex = 0;
+
+            for (int i = 0; i < uvs.Count;)
+            {
+                Vector2 tmpUv = Vector2.zero;
+                tmpUv.x = uvs[i++];
+                tmpUv.y = uvs[i++];
+
+                uvsResult[uvsResultIndex++] = tmpUv;
+            }
```

Comment:
Not sure if this won't create some unnecessary defensive copies of the `Vector2` struct.
maybe just in case it might be better just to create the vector when assigning it to the result ` uvsResult[uvsResultIndex++] = new Vector2(uv1,uv2);`?

Reply:
This should be fixed from the Unity from 2 or 3 years If I remember correctly, but I will change it just in case